aws/main.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/vault/api"
+)
+
+func main() {
+	config := api.DefaultConfig()
+	vaultClient, err := api.NewClient(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	content, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	s, err := vaultClient.Logical().Write("/auth/kubernetes/login", map[string]interface{}{
+		"role": "demo",
+		"jwt":  string(content[:]),
+	})
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	log.Println("==> WARNING: Don't ever write secrets to logs.")
+	log.Println("==>          This is for demonstration only.")
+	log.Printf("Vault token: %s\n", s.Auth.ClientToken)
+
+	vaultClient.SetToken(s.Auth.ClientToken)
+	s, err = vaultClient.Logical().Read("/aws/creds/readonly")
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	// Give some time for IAM cred creation to propagate since this action is
+	// eventually consistent
+	time.Sleep(30 * time.Second)
+
+	accessKey := s.Data["access_key"].(string)
+	secretKey := s.Data["secret_key"].(string)
+	log.Printf("AWS Access Key: %s\n", accessKey)
```

Comment:
Just for general safety, can we add the same "don't print these" warning as above here?